app/controllers/passengers_controller.rb

```diff
@@ -0,0 +1,66 @@
+class PassengersController < ApplicationController
+  def index
+    @passengers = Passenger.all
+  end
+
+  def show
+    passenger_id = params[:id].to_i
+    @passenger = Passenger.find_by(id: passenger_id)
+
+    if @passenger.nil?
+      head :not_found
+    end
+  end
+
+  def new
+    @passenger = Passenger.new
+  end
+
+  def create
+    @passenger = Passenger.new(passenger_params)
+
+    if @passenger.save
+      redirect_to passenger_path(@passenger.id)
+    else
+      redirect_to passengers_path
+    end
+  end
+
+  def edit
+    @passenger = Passenger.find_by id: params[:id]
+    if @passenger.nil?
+      redirect_to root_path
+    end
+  end
+
+  def update
+    @passenger = Passenger.find_by id: params[:id]
+    if @passenger.nil?
+      head :not_found
+      return
+    elsif @passenger.update(passenger_params)
+      redirect_to passenger_path(@passenger.id)
+      return
+    else
+      render :edit
```

Comment:
You should also include a status code

```suggestion
      render :edit, status: :bad_request
```